### PR TITLE
Platform separation

### PR DESCRIPTION
**Pull request by carstene1ns**
Tried to clean up player main code base.

### CODE REVIEW

**Comment by Ghabry** [comment]
only quickly scrolled through it: I like this approach, especially the early setup with the individual mains.

File: src/player.h [hunk 1]
```diff
@@ -59,7 +59,7 @@ namespace Player {
 	/**
 	 * Initializes EasyRPG Player.
 	 */
-	void Init(int argc, char *argv[]);
+	void Init(std::vector<std::string> arguments);
```
Comment (Ghabry) [inline src/player.h:62]: nit: Here and below: Pass as ``const`` reference ``&``

Comment (carstene1ns) [reply src/player.h:62]: Actually wanted to ask if this would be a good idea to `std::move` the vector.

**Comment by Ghabry** [comment]
this looks good so far

File: src/platform/sdl/sdl_ui.cpp [hunk 2]
```diff
@@ -20,26 +20,20 @@
...
 #include "color.h"
 #include "graphics.h"
 #include "keys.h"
 #include "output.h"
 #include "player.h"
 #include "bitmap.h"
 
-#include "audio.h"
+#ifdef GEKKO
+  #include "platform/wii/main.h"
```
Comment (fdelapena) [inline src/platform/sdl/sdl_ui.cpp:32]: nested starting #

File: src/player.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ namespace Player {
 	/**
 	 * Initializes EasyRPG Player.
```
Comment (fdelapena) [inline src/player.h:60]: Nit: there's a good chance for adding updated Doxygen lines here and for Game_Config

**Review by fdelapena — APPROVED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -960,13 +965,13 @@ if(${PLAYER_BUILD_EXECUTABLE} AND ${PLAYER_TARGET_PLATFORM} STREQUAL "SDL2")
 	if(APPLE)
 		set(EXE_NAME "EasyRPG-Player.app")
 		set_source_files_properties(${${PROJECT_NAME}_BUNDLE_ICON} PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
-		add_executable("${EXE_NAME}" MACOSX_BUNDLE "src/main.cpp" ${${PROJECT_NAME}_BUNDLE_ICON})
+		add_executable(${EXE_NAME} MACOSX_BUNDLE "src/platform/sdl/main.cpp" ${${PROJECT_NAME}_BUNDLE_ICON})
 		configure_file(resources/osx/Info.plist.in resources/osx/Info.plist @ONLY)
```
Comment (fdelapena) [inline CMakeLists.txt:969]: Nit: a good chance for renaming `osx` to `macos`.

**Comment by carstene1ns** [comment]
ready

**Review by Ghabry — APPROVED** [review]
Great. A totally reasonable refactor finally possible after getting rid of all the Makefiles.
Will also make adding new ports a much cleaner approach 👍

**Pull request merged**